.changeset/cold-snakes-train.md

```diff
@@ -0,0 +1,5 @@
+---
+"@astrojs/markdown-remark": minor
+---
+
+Adds a `data-language` attribute on the rendered `pre` and `code` elements to expose the highlighted syntax language
```

Comment:
``````suggestion
Adds a `data-language` attribute on the rendered `pre` and `code` elements to expose the highlighted syntax language.

This allows retrieving the language in a rehype plugin:

```js
// myRehypePre.js
import { visit } from "unist-util-visit";
export default function myRehypePre() {
  return (tree) => {
    visit(tree, { tagName: "pre" }, (node) => {
      const lang = node.properties.dataLanguage;
      [...]
    });
  };
}
```
``````

Hi @604qgc ! I was thinking of including something like this in the changeset, since for minor releases of new features we do like to show them off a bit and really help people understand why your thing is cool! :nail_care: 

I took this from your PR description, but if there's something else you'd prefer to use, or would change this in some way, please do! I think a changeset of this style/length is helpful to people to understand the new feature, so anything like this you can suggest would be great.